stress/shell/src/main/java/alluxio/stress/cli/MultiOperationStressMasterBench.java

```diff
@@ -0,0 +1,479 @@
+
+          // track max response time
+          long[] maxResponseTimeNs = mResult.getStatistics(operationIndex).mMaxResponseTimeNs;
+          int bucket =
+              Math.min(maxResponseTimeNs.length - 1, (int) ((currentMs - recordMs) / bucketSize));
+          if (responseTimeNs > maxResponseTimeNs[bucket]) {
+            maxResponseTimeNs[bucket] = responseTimeNs;
+          }
+        }
+      }
+    }
+
+    //Each thread will randomly pick an operation to do by operations ratio.
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runOperationsRatioBased() throws Exception {
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mOperationsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + mParameters.mOperationsRatio[i] / ratioSum;
+        last = ratioArray[i];
+      }
+      runInternal(lastOperationIndex -> {
+        double r = mRandom.nextDouble();
+        int operationIndex = 0;
+        for (int i = 0; i < mParameters.mOperations.length; i++) {
+          if (ratioArray[i] >= r) {
+            operationIndex = i;
+            break;
+          }
+        }
+        mContext.getRateLimiter(0).acquire();
+        return operationIndex;
+      });
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runThreadsRatioBased() throws Exception {
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mThreadsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + (mParameters.mThreadsRatio[i] / ratioSum) * mParameters.mThreads;
+        last = ratioArray[i];
+      }
+      int operationIndex = 0;
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        if (ratioArray[i] >= mThreadIndex) {
+          operationIndex = i;
+          break;
+        }
+      }
+      int finalOperationIndex = operationIndex;
+      runInternal(lastOperationIndex -> {
+        mContext.getRateLimiter(finalOperationIndex).acquire();
+        return finalOperationIndex;
+      });
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runInternalTargetThroughputBased() throws Exception {
+      RateLimiter[] rls = mContext.getRateLimiters();
+      runInternal((lastOperationIndex) -> {
+        int operationIndex = (lastOperationIndex + 1) % mParameters.mOperations.length;
+        mContext.getGrandRateLimiter().acquire();
+        while (!rls[operationIndex].tryAcquire()) {
+          operationIndex++;
+          operationIndex %= mParameters.mOperations.length;
+        }
+        return operationIndex;
```

Comment:
add some comments on how rate limiting on the global level and each op level works.

Reply:
done